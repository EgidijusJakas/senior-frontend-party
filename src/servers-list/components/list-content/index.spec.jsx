import React from 'react';
import { shallow } from 'enzyme';
import { ListContent } from '.';

describe('List Content', () => {
    it('should render loader', () => {
        const props = {
            serversList: [],
            loading: true
        };

        const renderedComponent = shallow(<ListContent {...props} />);
        expect(renderedComponent).toMatchSnapshot();
    });

    it('should render servers list', () => {
        const props = {
            serversList: [{
                distance: 11,
                name: 'Server1',
                key: '1'
            }, {
                distance: 1,
                name: 'Server2',
                key: '2'
            }],
            loading: false
        };

        const renderedComponent = shallow(<ListContent {...props} />);
        expect(renderedComponent).toMatchSnapshot();
    });

    it('should render error message', () => {
        const props = {
            serversList: [],
            loading: false,
            failedToLoad: true
        };

        const renderedComponent = shallow(<ListContent {...props} />);
        expect(renderedComponent).toMatchSnapshot();
    });
});
