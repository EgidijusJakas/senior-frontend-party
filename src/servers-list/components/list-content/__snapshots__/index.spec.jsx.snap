// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Content should render error message 1`] = `
<styled.div
  className="alert alert-danger"
>
  <FormattedMessage
    defaultMessage="Something went wrong. Failed to load servers list."
    id="src.servers-list.components.list-content.failedToLoadServersList"
    values={Object {}}
  />
</styled.div>
`;

exports[`List Content should render loader 1`] = `
<styled.div>
  <Component
    component="div"
    loaded={false}
    loadedClassName="loadedContent"
    parentClassName="loader"
  />
</styled.div>
`;

exports[`List Content should render servers list 1`] = `
Array [
  <styled.li
    className="list-group-item"
    key="1"
  >
    <span>
      Server1
    </span>
    <span>
      11 km
    </span>
  </styled.li>,
  <styled.li
    className="list-group-item"
    key="2"
  >
    <span>
      Server2
    </span>
    <span>
      1 km
    </span>
  </styled.li>,
]
`;
